### ISSUE TITLE
Allow all UTF-8 characters

### ISSUE THREAD

**Issue by gsteimer** [tsgrp/ActiveWizard#256]
Currently, only normal keyboard characters are supported in the HTML5 Active Form module.  On edge2 there's a TS Non-English form.  Here's what I'm seeing:

First time through - things look good:

_Enter text, go forward then back_
![image](https://cloud.githubusercontent.com/assets/1377228/9470638/6d2e56f6-4b12-11e5-9745-06aa8d11b96c.png)

_summary page_
![image](https://cloud.githubusercontent.com/assets/1377228/9470643/78718150-4b12-11e5-9ce4-bd300efa5453.png)

_PDF_
![image](https://cloud.githubusercontent.com/assets/1377228/9470650/84a76fac-4b12-11e5-9130-3b1114ba0763.png)

However - after editing this form, issues start arising:

_Go back to page_
![image](https://cloud.githubusercontent.com/assets/1377228/9470665/9d7c79fa-4b12-11e5-89d0-4e345bc9b925.png)

_Summary page (no changes made)_
![image](https://cloud.githubusercontent.com/assets/1377228/9470676/a82df400-4b12-11e5-83ce-ddcb3119b65e.png)

_PDF (no changes made)_
![image](https://cloud.githubusercontent.com/assets/1377228/9470685/b59476b4-4b12-11e5-9048-1dbff3d2d76c.png)

Side note - the active wizard admin is completely fubared when it comes to non-english characters.  I propose we don't do anything about that and hook into https://github.com/tsgrp/hpi/issues/674 when it is finished.  Basically, we would do this:
1. Only run AW Admin in standard ASCII characters
2. When you enter text for a form question or answer, use a key that points to a localized string in L10N.  Something like `${myform.myquestion.text}`, `${myform.myquestion.answerone}`, etc
3. In the HTML5 component, when it encounters text that has the `${something}` token, use HPI's Localization module to get the text.

**Comment by mikeblum** [comment]
I believe the root of the problem is how we are encoding the JSON string as it's being sent to OC/retrieved by the content/content call upon editing a saved  form. I'd be curious to see the raw JSON.

**Comment by mikeblum** [comment]
![utf_encoding_tomcat](https://cloud.githubusercontent.com/assets/3905463/10430547/7134d9d6-70c5-11e5-823c-26ff44d3cfbd.jpg)

from @ikaikajcorrea - it looks like we just need to configure our tomcat.

**Comment by ikaikajcorrea** [comment]
Adding the URIEncoding parameter to our tomcat/conf/server.xml configuration allowed for our client's Search Portal to search against Japanese characters.

**Comment by gsteimer** [comment]
I just tried this on edge1's tomcat, but it's not working - it still garbles the characters after editing the form:

![image](https://cloud.githubusercontent.com/assets/1377228/10437929/18ade7d6-70f6-11e5-8de8-5392b85e6e77.png)

Based on this doc: http://wiki.apache.org/tomcat/FAQ/CharacterEncoding, I think the problem is that the `URIEncoding` setting only affects `GET` requests.  The next thing to try based on the doc is to implement the filter to process `POST` requests with UTF-8.

**Comment by mikeblum** [comment]
updated the HPI wiki accordingly:

https://github.com/tsgrp/hpi/wiki/Installation-guide#utf-8-encoding

Issue remains with Asiatic (Korean, Chinese, Japanese) languages in Wizard Forms. The issue lies with how iText XMLWorker processes XML/HTML.

**Comment by gsteimer** [comment]
@mikeblum - this is not fixed, even for non-asiatic languages.  Here are some issues:

After editing the form, the accents are garbled in ActiveForm as well as the PDF:

![image](https://cloud.githubusercontent.com/assets/1377228/10544948/c889c042-73eb-11e5-8eb0-904759b92384.png)

![image](https://cloud.githubusercontent.com/assets/1377228/10544973/ed561308-73eb-11e5-965a-cb699597c34a.png)

This was tested on edge1 after the updates to the `server.xml` and `web.xml` mentioned in the new wiki entry.

**Comment by gsteimer** [comment]
Also - something really odd related to accented characters in the actual form (not the answers).

The first time though, the spanish question on page 2 looks like this:
![image](https://cloud.githubusercontent.com/assets/1377228/10545161/ea8f6d30-73ec-11e5-93ac-cca9d5b04a7f.png)

And the PDF:

![image](https://cloud.githubusercontent.com/assets/1377228/10545178/ff4b7390-73ec-11e5-958d-be3977e260a3.png)

BUT - after editing the form, it looks good:
Form:
![image](https://cloud.githubusercontent.com/assets/1377228/10545187/09d2cf0c-73ed-11e5-9c89-47b6961d5643.png)

PDF:
![image](https://cloud.githubusercontent.com/assets/1377228/10545193/110c61de-73ed-11e5-8f3f-658bc77f27a8.png)

**Comment by ZachGerth** [comment]
In https://github.com/tsgrp/hpi/issues/932, we weren't actually ensuring the JSON was in UTF-8 format before reading in from the byte stream - if we specified UTF-8, it fixed the issue where comments with non-UTF-8 characters were breaking the parsing - Java tries to find the UTF-8 equivalent, if one existed. This may be applicable to whoever looks into this issue later down the line.

**Comment by mwojno-tsg** [comment]
This issue was fixed within OC commit #25373 and AW commit #7807. 
- For AW, `<%@ page contentType="text/html; charset=UTF-8" %>` was added to the headers of `.jsp` files that were missing it. When you add a page, question, or answer now it should always be encoded to `utf-8` , which is the expected encoding across AW. 

- For OC, wizard specific code of `PDFUtil` and `FinishAndSubmitActionExecuter` was changed to have the `.getBytes("UTF-8")`. This was needed because whenever we are dealing with a string of the content we need to make sure that Java doesn't default encode to something other than `utf-8` as that string is written back to an input stream or something.

- In summary, we should now be able to confidently say we support end to end `utf-8` encoding/decoding for the wizard.
